modules/core/src/lib/mode-handlers/composite-mode-handler.js

```diff
@@ -0,0 +1,84 @@
+// @flow
+
+import type { FeatureCollection, Feature, Position } from '../../geojson-types.js';
+import type {
+  ClickEvent,
+  PointerMoveEvent,
+  StartDraggingEvent,
+  StopDraggingEvent
+} from '../event-types.js';
+import { ModeHandler, type EditAction, type EditHandle } from './mode-handler.js';
+
+export class CompositeModeHandler extends ModeHandler {
+  handlers: Array<ModeHandler>;
+  options: Object;
+
+  constructor(handlers: Array<ModeHandler>, options: Object = {}) {
+    super();
+    this.handlers = handlers;
+    this.options = options;
+  }
+
+  _coalesce<T>(callback: any => T, resultEval: ?(T) => boolean = null): T {
+    let result: T;
+
+    for (let i = 0; i < this.handlers.length; i++) {
+      result = callback(this.handlers[i]);
+      if (resultEval ? resultEval(result) : result) {
+        break;
+      }
+    }
+
+    return (result: any);
+  }
+
+  setFeatureCollection(featureCollection: FeatureCollection): void {
+    this.handlers.forEach(handler => handler.setFeatureCollection(featureCollection));
+  }
+
+  setModeConfig(modeConfig: any): void {
+    this.handlers.forEach(handler => handler.setModeConfig(modeConfig));
+  }
+
+  setSelectedFeatureIndexes(indexes: number[]): void {
+    this.handlers.forEach(handler => handler.setSelectedFeatureIndexes(indexes));
+  }
+
+  setDeckGlContext(context: Object) {
+    this.handlers.forEach(handler => handler.setDeckGlContext(context));
+  }
+
+  handleClick(event: ClickEvent): ?EditAction {
+    return this._coalesce(handler => handler.handleClick(event));
+  }
+
+  handlePointerMove(event: PointerMoveEvent): { editAction: ?EditAction, cancelMapPan: boolean } {
+    return this._coalesce(
+      handler => handler.handlePointerMove(event),
+      result => result && result.editAction
+    );
+  }
+
+  handleStartDragging(event: StartDraggingEvent): ?EditAction {
+    return this._coalesce(handler => handler.handleStartDragging(event));
+  }
+
+  handleStopDragging(event: StopDraggingEvent): ?EditAction {
+    return this._coalesce(handler => handler.handleStopDragging(event));
+  }
+
+  getTentativeFeature(): ?Feature {
+    return this._coalesce(handler => handler.getTentativeFeature());
+  }
+
+  getEditHandles(picks?: Array<Object>, groundCoords?: Position): EditHandle[] {
```

Comment:
how about unioning the edit handles?

Reply:
This will break the comparison `editHandles !== this.state.editHandles`
So I'm adding a todo for now.